.github/workflows/run-typecheck.yml

```diff
@@ -0,0 +1,13 @@
+name: run-typecheck
```

Comment:
I suggest putting these (run-typecheck and run-eslint) into the same workflow file (call it ci-checks or something) so that we have one line item in the Actions tab in github.  @frasercl @ShrimpCryptid do you have a strong preference for two workflow runs per check, or one?   (also I suggest running it on every push to a pull request, not sure if that is what this does)

Reply:
Yes, it currently will re-run every time the branch is pushed to!

I don't mind merging the two actions back into one. I originally split them into two different actions so that `eslint` failing wouldn't stop `typeCheck` from running, but looking at documentation it seems like I can have both in the same job and run even if a previous step failed.

Reply:
separate jobs, separate steps, or separate command lines in the same step are all ok by me.  But I think it's slightly better to have them both in one file just to keep the Actions log a little bit cleaner.